sycl/source/detail/memory_manager.cpp

```diff
@@ -110,6 +110,23 @@ void *MemoryManager::allocateMemBuffer(ContextImplPtr TargetContext,
   return NewMem;
 }
 
+void *MemoryManager::createSubBuffer(RT::PiMem ParentMem, size_t ElemSize,
+                                     id<3> Offset, range<3> Range,
+                                     std::vector<RT::PiEvent> DepEvents,
+                                     RT::PiEvent &OutEvent) {
+  waitForEvents(DepEvents);
+  OutEvent = nullptr;
+
+  RT::PiResult Error = PI_SUCCESS;
+  cl_buffer_region Region{Offset[0] * ElemSize, Range[0] * ElemSize};
```

Comment:
is it worth adding a TODO as a reminder to change this to a pi_buffer_region type, rather than a cl_buffer_region?